test/unit/tenant_test.rb

```diff
@@ -519,6 +519,206 @@
     end
   end
 
+  describe "cross-tenant associations" do
+    for_each_scenario do
+      setup do
+        ActiveRecord::Base.connection.add_column :announcements, :tenant_id, :string
+        ActiveRecord::Base.connection.add_column :announcements, :user_id, :integer
+
+        User.has_one :announcement
+        Announcement.belongs_to :user
+      end
+
+      test "has_one automatically scopes by tenant_id" do
+        TenantedApplicationRecord.create_tenant("foo") do
+          user = User.create!(email: "user@foo.example.org")
+          Announcement.create!(message: "Foo announcement", tenant_id: "foo", user: user)
```

Comment:
Ideally I'd like to be able to do this in an untenanted scope, but it looks like AR does some validation that requires a tenanted connection ...

I'd also like to be able to do this without explicitly setting the tenant id in Announcement, that is:

```ruby
Announcement.create! user: user
```

should set the tenant column automatically.